pkg/stanza/fileconsumer/internal/util/util.go

```diff
@@ -18,3 +18,7 @@ func MapCopy(m map[string]any) map[string]any {
 	}
 	return newMap
 }
+
+func Mod(x, y int) int {
+	return ((x % y) + y) % y
+}
```

Comment:
This is needed because in golang, the `%` operator acts as a remainder, not a typical math `modulus` and it can output return negative integers.

Reply:
My understanding is that it only returns negative if you apply it to a negative value, so can't we just avoid that and compute this inline?

```go
mostRecentIndex = (x - 1 + y) % y
```

or

```go
mostRecentIndex--
if mostRecentIndex < 0 {
  mostRecentIndex += t.pollsToArchive
}
```